packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -869,6 +871,15 @@ export interface SdkPackage<TServiceOperation extends SdkServiceOperation> {
   crossLanguagePackageId: string;
   namespaces: SdkNamespace<TServiceOperation>[];
   licenseInfo?: LicenseInfo;
+  metadata: {
+    /**
+     * The version of the package.
+     * If undefined, the package is not versioned.
+     * If `all`, the package is versioned with all versions.
```

Comment:
With our current assumption for Azure, should the value only be single api-version? So `all` may not happen in reality?